amf-aml/shared/src/main/scala/amf/aml/client/scala/AMLElementClient.scala

```diff
@@ -21,4 +28,103 @@ class AMLElementClient private[amf] (protected val configuration: AMLConfigurati
                                  references)
 
+  /**
+    * Get all instances of registered dialects
+    * @return a Seq of [[Dialect]]
+    */
+  def getDialects(): immutable.Seq[Dialect] = getDialectsByCondition(_ => true)
+
+  /**
+    * Get all names of registered dialects
+    * @return a Seq of names
+    */
+  def getDialectNames(): immutable.Seq[String] = getDialectsByCondition(_ => true).map(_.nameAndVersion())
+
+  /**
+    * Find an instance of registered dialect with the provided name and version
+    * @param name of the dialect to find
+    * @return a Seq of [[Dialect]]
+    */
+  def getDialect(name: String): immutable.Seq[Dialect] = getDialectsByCondition(dialectNameFilter(name))
+
+  /**
+    * Find an instance of registered dialect with the provided name and version
+    * @param name of the dialect to find
+    * @param version of dialect to find
+    * @return an Option of [[Dialect]]
+    */
+  def getDialect(name: String, version: String): Option[Dialect] =
+    getDialectsByCondition(dialectNameAndVersionFilter(name, version)).headOption
+
+  /**
+    * Get all instances of SemanticExtensions present in the registered dialects
+    * @return a Seq of [[SemanticExtension]]
+    */
+  def getExtensions(): immutable.Seq[SemanticExtension] = getDialects().flatMap(_.extensions())
+
+  /**
+    * Find all instances of semantic extensions in the registered dialects filtering by the param
+    * @param uri of the propertyTerm of the semantic extension to search
+    * @return a Map of [[Dialect]] to [[SemanticExtension]]
+    */
+  def findSemanticByPropertyTerm(uri: String): Map[Dialect, Seq[SemanticExtension]] =
+    getDialects().map(d => d -> findSemanticByPropertyTerm(d, uri)).toMap
+
+  /**
+    * Find all instances of semantic extensions in the provided dialect filtering by the param
+    * @param dialect where the semantic extension will be searched
+    * @param uri of the propertyTerm of the semantic extension to search
+    * @return a Seq of [[SemanticExtension]]
+    */
+  def findSemanticByPropertyTerm(dialect: Dialect, uri: String): Seq[SemanticExtension] =
+    SemanticExtensionHelper.byPropertyTerm(dialect).find(uri)
+
+  /**
+    * Find all instances of semantic extensions in the registered dialects filtering by the param
+    * @param uri of the target field of the semantic extension to search
+    * @return a Map of [[Dialect]] to [[SemanticExtension]]
+    */
+  def findSemanticByTarget(uri: String): Map[Dialect, Seq[SemanticExtension]] =
+    getDialects().map(d => d -> findSemanticByTarget(d, uri)).toMap
+
+  /**
+    * Find all instances of semantic extensions in the provided dialect filtering by the param
+    * @param dialect where the semantic extension will be searched
+    * @param uri of the target field of the semantic extension to search
+    * @return a Seq of [[SemanticExtension]]
+    */
+  def findSemanticByTarget(dialect: Dialect, uri: String): Seq[SemanticExtension] =
```

Comment:
Why would a user want to lookup the extensions for a specific dialect?

Reply:
Me imagine la situacion en la que por una razon de uso el config tenga mas n dialects y tengan definido por fuera de AMF que algunos aplican a un doc y otros no.
En todo caso lo voy a tener como metodo privado, asi que no vi razon para no exponerlo.

Reply:
Hmmm ese no es un feature que aplique a nosotros. Si no entiendo mal las semantic aplican a todos los documentos